### ISSUE TITLE
This issue is irrelevant now, please ignore.

### ISSUE THREAD

**Issue by ghost** [wavemotion-dave/NINTV-DS#4]
Sorry in advance if this isn't the right place to post this. I thought this was enough of a problem to warrant putting it here.

While this emulator is leagues above the official Intellivision Lives! DS emulator, it seems like the numberpad isn't the most responsive thing as sometimes it'll ignore inputs outright while menu options on the bottom screen still work fine. For some games, this won't affect you. However, games like Bomb Squad that have the numberpad as a **CRUCIAL** part of the experience are left near-unplayable unless you awkwardly assign each tool to a button on your DS.

I have already deduced that this is not due to my hardware. My New Nintendo 3DS XL doesn't have any issues with its touch screen and it has been calibrated correctly beforehand. I am running this on Twilight Menu++ and I have the latest version of NINTV-DS (3.5) as of writing this.

Is this something anyone else experiences, or is this a strange me-only occurrence? Because it's really making some games less than ideal to play (and I can play some of them properly on the Lives! collection but I honestly don't want to because this is better in general.)

EDIT: May just be that I didn't noticed that I had to hold the key down for a little bit before releasing. This may not have been much of an issue. [View my reply for more info]

**Comment by wavemotion-dave** [comment]
Thanks for the feedback - this is a perfect place to report such things.  

I honestly hadn't noticed any problems with the touch-pad... I tend to use custom overlays for many games that need them and I play a ton of Space Spartans, Treasure of Tarmin (which I play almost daily) and TRON Solar Sailor which both use the keypad a lot... and not experienced any issues. 

I will play with Bomb Squad more when I get home - and really go through the key detection algorithm with a fine-tooth comb to make sure it's working as expected. I can definitely improve this if it's a software issue (and it sounds like it is) - so rest assured that we'll get this right.

**Comment by ghost** [comment]
I've noticed JUST NOW that if I hold the position of where i'm tapping for a second and then release it works fine but it still feels weird compared to tapping the menu options because those are generally a bit snappier. Is that intended or not? I may have just gotten this wrong and it might not be an issue but I just found it weird compared to how Lives! on the DS works with its' lower screen touchpad.

Though I will say this, I have used the built-in overlays and have yet to use any "custom" overlays (though I might try that if they end up working better for whatever reason) so maybe that might be part of it? I don't know. Just thought it was worth sharing.

**Comment by wavemotion-dave** [comment]
I'll still take a look at it - I'd rather it be highly responsive.  I do have some code to "dampen down" the presses on the keypad so it takes a full press and it probably could stand some tweaking. At a minimum, I could also add a configuration item to increase the sensitivity / response time so it's more to people's liking.  Options are always good!

**Comment by ghost** [comment]
I do agree with making it something that can be adjusted in the options because there may be others who are okay with the way it is, so if you do end up working on something, by all means, try to make it an option if possible.

**Comment by wavemotion-dave** [comment]
So it turns out I'm at maximum sensitivity for both the virtual keypad - same sensitivity for the RESET/LOAD/CONFIG/etc. meta buttons.  I'm sampling 60 times per second (or every frame) - a reasonable response for a UI will do it at least 30 Hz (30 times per second) so I'm twice that.  I tried FWDIAG which registers key hits and it's very responsive - slight taps of the buttons will register fine. I tried BOMB SQUAD and that seemed fine to me as well.  I checked with one other guy who uses the emulator frequently and he hasn't seen any response issues at all.

One possibility... if you're running from a FlashCart, you are probably running in DS mode (slower CPU mode) and so games with intellivoice might not run at full speed. So maybe you're not getting the 60FPS and so you're also not getting the 60 samples per second of the touch pad. You can turn on FPS in Config -> Global Configuration and see what your frame rate might be. 

I also have no real feedback on a press... with a real keypad you would have the tactile feel and know that you've pressed it. I can at least add a key click - but I'd probably make it optional as some games (like the AD&D games) have key click feedback already and I'd hate to duplicate what's already being done by the game (it would sound jarring I imagine).

In short: I can slow down the response but I can't speed it up much (on the DSi, I might be able to sample 50% faster but honestly 60 times per second is pretty fast touch sampling).

**Comment by ghost** [comment]
I've had it on DSi mode AND it's been running at 60FPS, but since I'm getting quite used to it I'm starting to find this whole thread a bit irrelevant now so it's fine if you decide to close this issue.

**Comment by wavemotion-dave** [comment]
No worries - I do appreciate the feedback in any case!
